### ISSUE TITLE
[C++][Pistache-server] Model file is not generated if its defintion is stored in an external yaml file

### ISSUE THREAD

**Issue by CyrilleBenard** [OpenAPITools/openapi-generator#773]
##### Description

The generator does not produce the corresponding model file in case the "type" is defined, not in the "main" YAML file but outside in an external YAML file.

The include file directive is well generated in the source file but the header file does not exist so that the compilation failed.

The compilation error looks like :

```
g++ -c  -I./api -I./model -I./impl -Wall -g -std=c++11 -o obj/api/Stair1Api.o api/Stair1Api.cpp
In file included from api/Stair1Api.cpp:13:0:
api/Stair1Api.h:29:28: fatal error: ProblemDetails.h: No such file or directory
compilation terminated.
```

**NOTE**: If the content of the external YAML file is inserted in the main YAML file, the issue disappears. See comment in the YAML file below

##### openapi-generator version

Current master : 3.2.1-SNAPSHOT

##### OpenAPI declaration file content or url

Main YAML file **check_include_ProblemDetails.yaml**
```yaml
openapi: 3.0.0
info:
  version: 1.0.0
  title: Check generation of ProblemDetails include file
  description: A simple API to illustrate OpenAPI concepts

servers:
  - url: http://localhost:8080

paths:
  /stair1:
    post:
      summary: blabla
      operationId: check_generation
      tags:
        - Stair1
      requestBody:
        content:
          application/json:
            schema:
              type: string
        required: true
      responses:
        '200':
          description: Expected response to a valid request
          content:
            application/json:
              schema:
                type: string
        '403':
          $ref: './check_include_ProblemDetails_data.yaml#/components/responses/403'
        default:
          description: Unexpected error

# Below the content of check_include_ProblemDetails_data.yaml
# If these lines are uncommented, the issue disappears because the generator produces the ProblemDetails model files
#components:
#  schemas:
#    ProblemDetails:
#      type: object
#      properties:
#        type:
#          type: string
#        title:
#          type: string
#
#  responses:
#    '403':
#      description: Forbidden
#      content:
#        application/problem+json:
#          schema:
#            $ref: '#/components/schemas/ProblemDetails'
#    default:
#      description: Generic Error

```

External YAML file named **check_include_ProblemDetails_data.yaml**
```yaml

components:
  schemas:
    ProblemDetails:
      type: object
      properties:
        type:
          type: string
        title:
          type: string

  responses:
    '403':
      description: Forbidden
      content:
        application/problem+json:
          schema:
            $ref: '#/components/schemas/ProblemDetails'
    default:
      description: Generic Error
```

##### Command line used for generation

Generate :
```
openapi-generator-cli.sh generate -i ./check_include_ProblemDetails.yaml -g cpp-pistache-server -c ./config.json -o .
```

Compile :
```
g++ -c  -I./api -I./model -I./impl -Wall -g -std=c++11 -o obj/api/Stair1Api.o api/Stair1Api.cpp
```

##### Steps to reproduce

Generate & compile

##### Related issues/PRs
N/A

**Comment by tnmtechnologies** [comment]
Same issue occurs with the jaxrs-resteasy generator.

**Comment by CyrilleBenard** [comment]
Good surprise for me today : This bug has already been fixed at least since the current 3.3.2-SNAPSHOT branch. I did not notice from which release it has been fixed but it has.
